Sources/LeafKit/LeafSyntax.swift

```diff
@@ -291,3 +291,112 @@ extension Syntax: CustomStringConvertible {
         }
     }
 }
+
+extension Syntax {
+    mutating func inlineRefs(_ externals: [String: LeafAST], _ new: inout Syntax?) -> Set<String> {
+        var changed = false
+        var rf = Set<String>()
+
+        switch self {
+            case .conditional(let co):
+                var tail: Syntax.Conditional? = co
+                while tail != nil {
```

Comment:
Change the `var tail` to name it `currentTail`, then use `while let tail = currentTail {` as the loop condition, then you can get rid of all the `!` uses in the loop body and change the end to `currentTail = tail.next`.

Reply:
Done